bin/lint_commit_msg.py

```diff
@@ -104,6 +105,19 @@ def lint_commit_msg(commit_msg_file, editor):
         print ''
         print '\n'.join(errors)
         print ''
+        if not editor:
+            print (
+                '\n'.join(textwrap.wrap(
+                    'You do not have EDITOR set in your environment. Please '
+                    'set EDITOR in your environment if you want to be able '
+                    'to edit the commit message. In the meantime, you can do:'
+                )) +
+                '\n\n'
+                'git commit --amend\n\n'
+                'to fix issues.'
+            )
+            return 0
```

Comment:
I hadn't understood what you had meant when you told on #input - "We will treat it as if the user had entered 'i' ". Now after looking at the code I do :-)

I would say that we should not return 0 and let the commit succeed if EDITOR is not set. This is because we check the presence of EDITOR environment variable only if there are issues with the commit message. 

Instead we could prompt the user with the y/n/i choice and let him choose what to do. We can also enhance the description of the "i" option to say "the commit will succeed as is, I will edit the commit message manually if needed" and that of "n" option to say "the commit will fail, I will redo it"